packages/xo-web/src/xo-app/host/tab-storage.js

```diff
@@ -115,23 +115,26 @@ export default connectStore(() => {
   const srs = createGetObjectsOfType('SR').pick(createSelector(pbds, pbds => map(pbds, pbd => pbd.SR)))
 
   const storages = createSelector(pbds, srs, (pbds, srs) =>
-    map(pbds, pbd => {
-      const sr = srs[pbd.SR]
-      const { physical_usage: usage, size } = sr
+    map(
+      filter(pbds, pbd => srs[pbd.SR] !== undefined),
+      pbd => {
+        const sr = srs[pbd.SR]
```

Comment:
What do you think about using `_.compact` & add 
```js
 if (sr === undefined) return
```
?

Reply:
I prefer to filter out unwanted upstream values to avoid iterating over them during the `map` iteration.
it reduces the total number of iterations.